src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -1248,10 +1252,62 @@ export class ResultsViewPageStore {
             })
         }
     }, []);
+    
+    readonly virtualStudies = remoteData(sessionServiceClient.getUserVirtualStudies(), []);
+    
+    readonly virtualStudyIdToStudy = remoteData({
+        await: ()=>[this.virtualStudies],
+        invoke: async ()=>{
+            return _.keyBy(
+                this.virtualStudies.result.map(virtualStudy=>{
+                    let study = {
+                        allSampleCount:_.sumBy(virtualStudy.data.studies, study=>study.samples.length),
+                        studyId: virtualStudy.id,
+                        name: virtualStudy.data.name,
+                        description: virtualStudy.data.description,
+                        cancerTypeId: "My Virtual Studies"
+                    } as CancerStudy;
+                    return study;
+                }), x =>x.studyId);
+        }
+    },{});
+
+    //this is only required to show study name and description on the results page
+    readonly queriedStudies = remoteData({
+		await: ()=>[this.studyIdToStudy, this.virtualStudyIdToStudy],
+		invoke: async ()=>{
+            const queriedStudies = [];
```

Comment:
i would take this and put it into a helper file.  this function should be pure and accept as arguments all the instance properties it uses (i.e. this references)